controllers/argocd/argocd_controller.go

```diff
@@ -1,3 +1,19 @@
+/*
```

Comment:
The `Reconcile` function in this file still has the old signature `Reconcile(reconcile.Request) (reconcile.Result, error)`.  Shouldn't it have the new signature `Reconcile(context.Context, reconcile.Request) (reconcile.Result, error)` that was generated by `operator-sdk create api ...`?

Reply:
I thought these signatures were just a recommendation and retained the old conventions. Will update to use the latest signatures

Reply:
@jopit The new signature was introduced by `controller-runtime` v0.7.0. The PR depends on v0.0.16 of argocd-operator which uses the old `controller-runtime` v0.6.0. I couldn't update the controller-runtime only in the downstream operator. We could either 
1. wait for the argocd-operator to have the updated controller-runtime and then pull the version downstream
2. Or get this PR merged with old signatures and then change them in a follow up PR with the updated upstream operator version

WDYT?